components/feature/downloads/src/main/java/mozilla/components/feature/downloads/DownloadsFeature.kt

```diff
@@ -210,6 +270,47 @@ class DownloadsFeature(
         block(Pair(state, download))
     }
 
+    /**
+     * Find all the app that can perform a download, including this app.
+     */
+    @VisibleForTesting
+    internal fun getDownloaderApps(context: Context, download: DownloadState): List<DownloaderApp> {
+        val downloadIntent = download.toIntent()
+
+        // Search for all activities that can render HTML.
+        val browsersIntent = Intent.parseUri("http://www.example.com/index.html", Intent.URI_INTENT_SCHEME)
+        val browsers = context.packageManager.queryIntentActivities(browsersIntent, 0)
+                .orEmpty()
+                .associateBy { it.activityInfo.name }
+
+        val thisApp = browsers.values
+                .firstOrNull { it.activityInfo.packageName == context.packageName }
+                ?.toDownloaderApp(context, download)
+
+        // Return only the apps that can perform a download plus this app.
+        return context.packageManager.queryIntentActivities(downloadIntent, 0)
+                .filter { !browsers.contains(it.activityInfo.name) ||
+                        it.activityInfo.packageName == context.packageName }
+                .map { it.toDownloaderApp(context, download) } + listOfNotNull(thisApp)
```

Comment:
Such code exists in multiple places: Here, `feature-applinks`, `Browsers` in `support-utils`. Eventually we may want to share that better. Maybe also something to just file for now?

Reply:
Agree, while coding I took a look to `Browsers` there are some differences, but we should find a way to share as much as possible. I will take a look to the other component to see if the change can be done here, if not I will file a ticket

Reply:
I refactored the implementation to share the behaviour with `Browsers`,  for `feature-applinks` will required a bit more work, I filed https://github.com/mozilla-mobile/android-components/issues/8041 for it.